sys/net/application_layer/nanocoap/nanocoap.c

```diff
@@ -595,6 +604,80 @@ size_t coap_put_option_uri(uint8_t *buf, uint16_t lastonum, const char *uri, uin
     return bufpos - buf;
 }
 
+/* Common functionality for addition of an option */
+static ssize_t _add_opt_pkt(coap_pkt_t *pkt, uint16_t optnum, uint8_t *val,
+                            size_t val_len)
+{
+    uint16_t lastonum = (pkt->options_len)
+            ? pkt->options[pkt->options_len - 1].opt_num : 0;
+    assert(optnum >= lastonum);
+
```

Comment:
I think you should check here whether there is still an empty position in the options array, to prevent array-out-of-bounds bugs

Reply:
Excellent idea!